File: programming/lab2/src/main/java/moves/status/ThunderWave.java
package moves.status;

import ru.ifmo.se.pokemon.*;

/**
 * Paralyzes the target.
 */
public class ThunderWave extends StatusMove {

    public ThunderWave() {
        super(Type.ELECTRIC, 100, 90);
    }

    @Override
    protected String describe() {
        return "использует Thunder Wave";
    }

    @Override
    protected void applyOppEffects(Pokemon pokemon) {
        Effect.paralyze(pokemon);
    }
}

File: programming/lab2/app/src/main/kotlin/org/itmo/moves/status/ThunderWave.kt
package org.itmo.moves.status

import ru.ifmo.se.pokemon.Effect
import ru.ifmo.se.pokemon.Pokemon
import ru.ifmo.se.pokemon.StatusMove
import ru.ifmo.se.pokemon.Type

class ThunderWave : StatusMove(Type.ELECTRIC, 0.0, 90.0) {
    override fun applyOppEffects(opp: Pokemon?) {
        Effect.paralyze(opp);
    }

    override fun describe(): String {
        return "использует Thunder Wave"
    }
}